File: lemur-libs/lemur-testkit/src/main/java/io/github/malczuuu/lemur/testkit/archunit/NamingRules.java
package io.github.malczuuu.lemur.testkit.archunit;

import static com.tngtech.archunit.base.DescribedPredicate.describe;
import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;

import com.tngtech.archunit.lang.ArchRule;

public final class NamingRules {

  public static final ArchRule TEST_CLASSES_PLURAL =
      classes()
          .that()
          .containAnyMethodsThat(
              describe(
                  "are annotated with @Test or @ArchTest",
                  m ->
                      m.isAnnotatedWith(org.junit.jupiter.api.Test.class)
                          || m.isAnnotatedWith(com.tngtech.archunit.junit.ArchTest.class)))
          .should()
          .haveSimpleNameEndingWith("Tests");

  private NamingRules() {}
}

File: lemur-libs/lemur-testkit/src/main/kotlin/io/github/malczuuu/lemur/testkit/archunit/NamingRules.kt
package io.github.malczuuu.lemur.testkit.archunit

import com.tngtech.archunit.base.DescribedPredicate.describe
import com.tngtech.archunit.junit.ArchTest
import com.tngtech.archunit.lang.ArchRule
import com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes
import org.junit.jupiter.api.Test

object NamingRules {

    val TEST_CLASSES_PLURAL: ArchRule =
        classes()
            .that().containAnyMethodsThat(
                describe("are annotated with @Test or @ArchTest") {
                    it.isAnnotatedWith(Test::class.java) || it.isAnnotatedWith(ArchTest::class.java)
                },
            )
            .should().haveSimpleNameEndingWith("Tests")
}
